nexus/social_media/services.py

```diff
@@ -0,0 +1,166 @@
+# Third Party Stuff
+import requests
+from django.conf import settings
+from django.utils import timezone
+from rest_framework import status
+
+# nexus Stuff
+from nexus.base import exceptions as exc
+from nexus.social_media.models import Post
+
+
+def update_post_object(post):
```

Comment:
We don't need this function. We'll do all updates in `publish_posts_service`. We'll first update all those posts that are ready to get published and then will call functions to publish each post. If we update each post like this function is doing, we may break the working. Suppose there are 1000 posts needed to be published and you are publishing and updating each post, now you are done with 500 posts, and next celery beat get executed then the rest 500 posts that were not published and not updated will _again_ get in the async list.

Reply:
But there could be this case:- We update all the posts that are ready to be published but when we actually publish the post, the API may break. That would lead to inconsistent data as there would be post which aren't published but data would show that they have been published.

Reply:
In that case, we can update the `post` oppositely. Like removing its `is_published = true` to `false` and so on.